splice/ingest.py

```diff
@@ -487,3 +413,59 @@ def distribute(data, channel_id, deploy, scheduled_dt=None):
             insert_distribution(url, channel_id, deploy, scheduled_dt)
 
     return distributed
+
+
+def compress_payload(payload):
+    """ Compress payload by re-using image uri and enhanced image uri. Note
```

Comment:
How about:

Make an inplace modification to the payload that compresses it by creating an asset dictionary and referring to the assets by ID in the payload